gtsam/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
+project(gtsam LANGUAGES CXX)
```

Comment:
Just want to make sure I understand what you did here: We now have a top-level project `GTSAM`, and two sub-projects `gtsam` and `gtsam_unstable`, and the main benefit of having the sub-projects is that we don't have to write `/gtsam/`, or is there something else less obvious that I'm missing?

Reply:
Exactly, the top-level was named all upper case just to make it different from the actual library. 

I normally add these `project()`s for many subdirectories because it's handy: for some IDEs (e.g. MSVC) CMake generates "solutions" that only include the dependent projects, so for large projects it's much faster to load only that part of the project.

Anyway, that's **not** why I added the two projects: it was the only way I found to prevent MSVC to fail, it was done here: https://github.com/borglab/gtsam/commit/65f442ce53b4dd7dd29435e19eb1089895a1b235

Unfortunately, I didn't documented very well (at all, actually...) **which** was the error and can't remember it. Could checkout the former version just to test, will try to do it if time permits. 

Bottom line anyway: something in the targets configuration broke with the new variables and the only way to make CMake to get things right was to add these project() definitions.

@chrisbeall : What do you think of the new build CMake variables? I'm afraid you will think there are too many cache entries now, but in this way things should be much easier when packaging GTSAM as a Debian packages with CMake exported projects and such, for example.

Reply:
Ok, that makes sense. Definitely useful to add a comment for those kinds of workarounds in the future.
I think the new CMake variables are good, although I haven't yet tested building another project against it. 
Have you had a chance to test this on Windows? Seems we always get surprises there.